src/subscription_manager/repofile.py

```diff
@@ -461,19 +461,28 @@ def sections(self):
 
         def fix_content(self, content):
             # Luckily apt ignores all Fields it does not recognize
-            baseurl = content["baseurl"]
+            parsed_url = urlparse(unquote(content["baseurl"]))
+            baseurl = parsed_url._replace(query="").geturl()
             url_res = re.match(r"^https?://(?P<location>.*)$", baseurl)
             ent_res = re.match(r"^/etc/pki/entitlement/(?P<entitlement>.*).pem$", content["sslclientcert"])
             if url_res and ent_res:
                 location = url_res.group("location")
                 entitlement = ent_res.group("entitlement")
                 baseurl = "katello://{}@{}".format(entitlement, location)
 
+            query = parse_qs(parsed_url.query)
+            if "rel" in query and "comp" in query:
+                suites = query["rel"][0].replace(",", " ")
+                components = query["comp"][0].replace(",", " ")
```

Comment:
is they are comma-separated lists, please split+join them rather than replace the separator; even if the likelyhood of mistakes here is small, I've bitten by mishappenings of "oh let's play with lists as strings" many times in the past...

Reply:
I will have a look into trying this out (likely early next week).

Reply:
This is what I came up with for using split+join: https://github.com/candlepin/subscription-manager/pull/3454/commits/c06c4b50671e01186500d2d75721799be662e688

Reply:
I adjusted these two lines to use spit and join instead.